GetStream/stream-chat-android
# Feature/3361 allow ignore mutes for unread count

### 🎯 Goal
Some of our customers want to support social media like use cases where muted channels receive unread count events.

This is required to support it. Fixes #3361 

### 🛠 Implementation details

Added a flag in the OP `Config` that affects the unread count logic.

### 🧪 Testing

1. Run the app on two devices on this branch with the applied patch.
2. Log in to two different users who are in the same channel.
3. Mute a channel with one user and send a message with another. The unread count should update.
4. You can also test by putting the flag to false (default behavior) that the unread count doesn't update when the channel is muted.

<details>
  <summary>Patch file to apply these changes for testing</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(revision 60665b59092afdb9b5238a1c7c7fe90401a0edae)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(date 1650444246174)
@@ -67,7 +67,7 @@
             config = Config(
                 userPresence = true,
                 persistenceEnabled = true,
-                ignoreMutesForUnreadCount = false
+                ignoreMutesForUnreadCount = true
             ),
             appContext = context
         )
Index: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt
--- a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt	(revision 60665b59092afdb9b5238a1c7c7fe90401a0edae)
+++ b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt	(date 1650440259516)
@@ -58,7 +58,7 @@
             config = Config(
                 userPresence = true,
                 persistenceEnabled = true,
-                ignoreMutesForUnreadCount = false
+                ignoreMutesForUnreadCount = true
             ),
             appContext = context
         )

```

</details>

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
